# Trim Galore: Trimming Our Sequences
Alright, y'all. I know what you're thinking. _Omg, Tammi, why would we trim our amazing reads?_ Well, the answer to that, silly, is that trimming our reads removes low-quality bases. We are all about that base here (high-five if you get that reference)! Anyways, it's very common that the end of reads have lower accuracy. It's similar to how I can't read at a 12th grade level for more than five minutes, some processes just are not entirely consistent.
> Trim Galore is a bioinformatics tool that automates the process of cleaning up raw sequencing data by performing both quality and adaptor trimming. It essentially removes unnecessary parts of the read before we put it through ruther anaysis.

Alright, so we will conduct our trimming using the script below:
* [trim.sh]()

This job is going to take a bit, so I have some things to tell you while you wait. Typically, people will fastqc this step as well, however, you remember how long that took last time, don't you? Because I am so fricking nice, I have decided to do that for you. I know, crazy. Anyways, if you are interested in looking at the fastqc reports that were generated by these trimmed reads, you can look at that [here]().

If you still have time you need to kill while waiting for the job to finish, might I suggest an activity: [click me](https://papas-freezeria.io/)!

Now that we have trimming done, we can move on to [Part V]()! 